core/transaction.go

```diff
@@ -57,15 +92,64 @@ type InvokeTransaction struct {
 	Version *felt.Felt
 }
 
-func (i *InvokeTransaction) Hash() *felt.Felt {
+func (i *InvokeTransaction) Hash(chainId []byte) (*felt.Felt, error) {
```

Comment:
We can shorten this function by doing something like this:
```go
func (it *InvokeTransaction) Hash(chainId *felt.Felt) (*felt.Felt, error) {
	invokeFelt := new(felt.Felt).SetBytes([]byte("invoke"))
	if it.Version.IsZero() {
		return crypto.PedersenArray(
			invokeFelt,
			it.Version,
			it.ContractAddress,
			it.EntryPointSelector,
			crypto.PedersenArray(it.CallData...),
			it.MaxFee,
			chainId,
		), nil
	} else if it.Version.IsOne() {
		return crypto.PedersenArray(
			invokeFelt,
			it.Version,
			it.SenderAddress,
			new(felt.Felt),
			crypto.PedersenArray(it.CallData...),
			it.MaxFee,
			chainId,
			it.Nonce,
		), nil
	}
	return nil, errors.New("invalid transaction version")
}
```

Reply:
Done